examples/pytorch/argo/ogb_example_ARGO.py

```diff
@@ -0,0 +1,349 @@
+    train_nid, val_nid, test_nid, in_feats, labels, n_classes, nfeat, g = data
+
+    # Create PyTorch DataLoader for constructing blocks
+    sampler = dgl.dataloading.MultiLayerNeighborSampler(
+        [int(fanout) for fanout in args.fan_out.split(",")]
+    )
+    dataloader = dgl.dataloading.DataLoader(
+        g,
+        train_nid,
+        sampler,
+        batch_size=b_size,
+        shuffle=True,
+        drop_last=False,
+        num_workers=len(load_core),
+        use_ddp=True,
+    )
+
+    # Define model and optimizer
+    model = SAGE(
+        in_feats,
+        args.num_hidden,
+        n_classes,
+        args.num_layers,
+        F.relu,
+        args.dropout,
+    )
+    model = model.to(device)
+    model = DistributedDataParallel(model)
+    loss_fcn = nn.CrossEntropyLoss()
+    optimizer = optim.Adam(model.parameters(), lr=args.lr, weight_decay=args.wd)
+
+    # Training loop
+    avg = 0
+    iter_tput = []
+    best_eval_acc = 0
+    best_test_acc = 0
+    PATH = "model.pt"
+    if counter[0] != 0:
+        checkpoint = th.load(PATH)
+        model.load_state_dict(checkpoint["model_state_dict"])
+        optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
+        epoch = checkpoint["epoch"]
+        loss = checkpoint["loss"]
+
+    with dataloader.enable_cpu_affinity(
+        loader_cores=load_core, compute_cores=comp_core
+    ):
+        for epoch in range(ep):
+            tic = time.time()
+
+            # Loop over the dataloader to sample the computation dependency graph as a list of
+            # blocks.
+            for step, (input_nodes, seeds, blocks) in enumerate(dataloader):
+                tic_step = time.time()
+
+                # copy block to gpu
+                blocks = [blk.int().to(device) for blk in blocks]
+
+                # Load the input features as well as output labels
+                batch_inputs, batch_labels = load_subtensor(
+                    nfeat, labels, seeds, input_nodes
+                )
+
+                # Compute loss and prediction
+                batch_pred = model(blocks, batch_inputs)
+                loss = loss_fcn(batch_pred, batch_labels)
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+
+                iter_tput.append(len(seeds) / (time.time() - tic_step))
+                if step % args.log_every == 0:
+                    acc = compute_acc(batch_pred, batch_labels)
+                    print(
+                        "Step {:05d} | Loss {:.4f} | Train Acc {:.4f} | Speed (samples/sec) {:.4f}".format(
+                            step,
+                            loss.item(),
+                            acc.item(),
+                            np.mean(iter_tput[3:]),
+                        )
+                    )
+
+            toc = time.time()
+            print("Epoch Time(s): {:.4f}".format(toc - tic))
+            if epoch >= 5 and rank == 0:
+                avg += toc - tic
```

Comment:
For first _n_search_ epochs it never comes here as number of epochs during tuning (_ep_) is always set to 1. 
This cause that final result will be not valid in my opinion.
I think  we have to measure total time of the script as most important for evaluating ARGO usability. 

Maybe additionally we can show "mean epoch time after tuning" as additional information - so here could be "_epoch_  >=1" , but later we have explicitly write that it is "mean epoch time after tuning" and of course divided _avg_ by adequate number of epochs

Reply:
Thanks. I made some changes, and now the program reports two values: (1) Avg epoch time and (2) Avg epoch time after auto-tuning.
I also changed the if-else condition to if epoch >= 1.